src/routes/safe/container/selector.ts

```diff
@@ -31,24 +32,31 @@ export const extendedSafeTokensSelector = createSelector(
   (safeBalances, tokensList, ethAsToken): List<Token> => {
     const extendedTokens: Array<Token> = []
 
-    safeBalances.forEach((safeBalance) => {
-      const tokenAddress = safeBalance.tokenAddress
+    try {
```

Comment:
I think a small `if` with logging + an early return would be better.

In a big try-catch like this, an error can happen on any line inside and we won't know exactly what broke. Try-catches should be as small as possible.

Reply:
Like you suggested initially:
```
if (!Array.isArray(safeBalances)) {
  Sentry.captureMessage(...)
  return List([])
}
```
WDYT?

Reply:
Yes, I changed to the try/catch approach because after that we mentioned that we could return an empty list at any failure case, but I'm also fine with checking just this specific case :+1:

Reply:
Thanks!

Since we know why exactly it was happening and can test it, the other code shouldn't break. And if it does for an unrelated reason, we'll see it in Sentry. 😄